### PR TITLE
Release/3.0.0

### PR DESCRIPTION
**Pull request by mvktsk**
#VP-636
2 new articles created:
* Deploy Platform on Windows
* Deploy Platform on Linux

Read.me edited:
* Text description how to deploy from precompiled binaries removed
* Links to "Deploy Platform on Windows" "Deploy Platform on Linux" added

### CODE REVIEW

File: docs/deploy-from-precompiled-binaries-linux.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+```console
+Now listening on: http://localhost:5000
+Now listening on: https://localhost:5001
+```
+
+### First sign in
+
+* Open in your browser follow url `https://localhost:5001` in the warning for not private connections that appears click advanced and continue work. How to remove this error and use a trusted self-signed certificate please read in this article [Trust the ASP.NET Core HTTPS development certificate](https://www.hanselman.com/blog/DevelopingLocallyWithASPNETCoreUnderHTTPSSSLAndSelfSignedCerts.aspx)
+* On the first request the application will create and initialize database. After that you should see the sign in page. Use the following credentials:
```
Comment (eMazeika) [inline docs/deploy-from-precompiled-binaries-linux.md]: The application will create and initialize database on the first request.

File: docs/deploy-from-precompiled-binaries-windows.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+---
+## Summary
+
+Use this guide to <a class="crosslink" href="https://virtocommerce.com/ecommerce-hosting" target="_blank">deploy</a> and configure precompiled Virto Commerce Platform V3.
+
+### Prerequisites
+
+[Prerequisites for .NET Core on Windows](https://docs.microsoft.com/en-us/dotnet/core/windows-prerequisites)
+
+[.NET Core SDK](https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-3.0.100-windows-x64-installer)
```
Comment (eMazeika) [inline docs/deploy-from-precompiled-binaries-windows.md]: 1. why SDK, isn't Runtime enough?
2. it's .NET 2.2 app. Why to point to specific ver. 3.0.100? Will it work?

File: docs/deploy-from-precompiled-binaries-windows.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+[Microsoft SQL Server](https://www.microsoft.com/en-us/sql-server/sql-server-2017-editions)
+
+## Downloading the precomplied binaries
+
+* Navigate to the <a href="https://github.com/VirtoCommerce/vc-platform/releases">Releases section of Virto Commerce Platform in GitHub.</a>
+
+* You will find **VirtoCommerce.Platform.3.x.x.zip** file. In this file the site has already been built and can be run without additional compilation. It does not includes all the source code.
```
Comment (eMazeika) [inline docs/deploy-from-precompiled-binaries-windows.md]: ... The source code is not included.

File: docs/deploy-from-precompiled-binaries-windows.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            "PublicUrl": "https://localhost:5001/assets/" <-- Set your platform application url with port localhost:5001
+        },
+    },
+```
+
+### Running the Platform by CLI "dotnet"
+
+* Install and trust HTTPS certificate
+
+To trust the .NET Core SDK HTTPS development certificate perform run:
```
Comment (eMazeika) [inline docs/deploy-from-precompiled-binaries-windows.md]: Run to trust the .NET Core SDK HTTPS development certificate:

File: docs/deploy-from-precompiled-binaries-windows.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+```console
+dotnet.exe dev-certs https --trust
+```
+
+Read more about [enforcing HTTPS in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-3.0&tabs=visual-studio#trust)
+
+* Run the Platform by follow command
```
Comment (eMazeika) [inline docs/deploy-from-precompiled-binaries-windows.md]: following command:

File: docs/deploy-from-precompiled-binaries-windows.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+```
+
+The output in the console will say something like:
+
+```console
+Now listening on: http://localhost:5000
+Now listening on: https://localhost:5001
+```
+
+### First sign in
```
Comment (eMazeika) [inline docs/deploy-from-precompiled-binaries-windows.md]: check Linux version

File: docs/deploy-from-precompiled-binaries-windows.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+* Open in your browser follow url `https://localhost:5001` in the warning for not private connections that appears click advanced and continue work. How to remove this error and use a trusted self-signed certificate please read in this article [Trust the ASP.NET Core HTTPS development certificate](https://www.hanselman.com/blog/DevelopingLocallyWithASPNETCoreUnderHTTPSSSLAndSelfSignedCerts.aspx)
+* On the first request the application will create and initialize database. After that you should see the sign in page. Use the following credentials:
+  * Login: **admin**
+  * Password: **store**
+
+### Host on Windows with IIS
+
+VirtoCommerce.Platform project already include the **web.config** file with all necessary settings for running in IIS.
+How to configure IIS application to host ASP.NET Core site please learn more in the official Microsoft ASP.NET Core documentation
+[Host ASP.NET Core on Windows with IIS](https://docs.microsoft.com/en-us/aspnet/core/publishing/iis)
```
Comment (eMazeika) [inline docs/deploy-from-precompiled-binaries-windows.md]: **web.config** file is already included in the release package. It contains all the necessary settings for running in IIS.
Read more in the official Microsoft ASP.NET Core documentation: [Host ASP.NET Core on Windows with IIS](https://docs.microsoft.com/en-us/aspnet/core/publishing/iis).

File: docs/deploy-from-precompiled-binaries-linux.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+### Running the Platform by CLI "dotnet"
+
+* Install and trust HTTPS certificate
+
+To trust the .NET Core SDK HTTPS development certificate on Linux run steps described in the [article](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-3.0&tabs=visual-studio#trust-https-certificate-from-windows-subsystem-for-linux)
+
+Read more about [enforcing HTTPS in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-3.0&tabs=visual-studio#trust)
+
+* Run the Platform by follow command
```
[the remark this answers is not in the document]
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-linux.md]: fixed

File: docs/deploy-from-precompiled-binaries-linux.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+* In the **ConnectionStrings** section change **VirtoCommerce** node (provided user should have permission to create new database):
+
+```json
+    "ConnectionStrings": {
+        "VirtoCommerce" : "Data Source={SQL Server URL};Initial Catalog={Database name};Persist Security Info=True;User ID={User name};Password={User password};MultipleActiveResultSets=True;Connect Timeout=30"
+    },
+
+```
+
+* In the **Assets** section set public url for assets `Assets:FileSystem:PublicUrl` with url of your application, this step is needed in order for display images
```
[the remark this answers is not in the document]
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-linux.md]: fixed

File: docs/deploy-from-precompiled-binaries-linux.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+            "PublicUrl": "https://localhost:5001/assets/" <-- Set your platform application url with port localhost:5001
+        },
+    },
+```
+
+### Running the Platform by CLI "dotnet"
+
+* Install and trust HTTPS certificate
+
+To trust the .NET Core SDK HTTPS development certificate on Linux run steps described in the [article](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-3.0&tabs=visual-studio#trust-https-certificate-from-windows-subsystem-for-linux)
```
[the remark this answers is not in the document]
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-linux.md]: fixed

File: docs/deploy-from-precompiled-binaries-linux.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+```
+
+The output in the console will say something like:
+
+```console
+Now listening on: http://localhost:5000
+Now listening on: https://localhost:5001
+```
+
+### First sign in
```
[the remark this answers is not in the document]
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-linux.md]: fixed

File: docs/deploy-from-precompiled-binaries-linux.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+
+* Open in your browser follow url `https://localhost:5001` in the warning for not private connections that appears click advanced and continue work. How to remove this error and use a trusted self-signed certificate please read in this article [Trust the ASP.NET Core HTTPS development certificate](https://www.hanselman.com/blog/DevelopingLocallyWithASPNETCoreUnderHTTPSSSLAndSelfSignedCerts.aspx)
```
[the remark this answers is not in the document]
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-linux.md]: fixed

File: docs/deploy-from-precompiled-binaries-linux.md (hunk 1 above)
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-linux.md]: fixed

File: docs/deploy-from-precompiled-binaries-windows.md (hunk 2 above)
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-windows.md]: 1. As i understend Runtime does not contain `dotnet` tool
2. Link to latest dotnet core SDK it works :)

File: docs/deploy-from-precompiled-binaries-windows.md (hunk 3 above)
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-windows.md]: Fixed

File: docs/deploy-from-precompiled-binaries-windows.md (hunk 4 above)
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-windows.md]: Fixed

File: docs/deploy-from-precompiled-binaries-windows.md (hunk 5 above)
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-windows.md]: fixed

File: docs/deploy-from-precompiled-binaries-windows.md (hunk 7 above)
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-windows.md]: Fixed

File: docs/connect-storefront-to-platform-v3.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
+# Connect Storefront to Platform
+
+* Deploy the latest storefront version as described [there](https://virtocommerce.com/docs/vc2devguide/deployment/storefront-deployment)
```
Comment (eMazeika) [inline docs/connect-storefront-to-platform-v3.md]: *Deploy the latest Storefront version as [described here](https://virtocommerce.com/docs/vc2devguide/deployment/storefront-deployment)

File: docs/connect-storefront-to-platform-v3.md [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+
+Note if you deploy the Storefront on local environment with the Platform you should host it with IIS
```
Comment (eMazeika) [inline docs/connect-storefront-to-platform-v3.md]: Note, if Platform and Storefront are deployed in the on-premises environment, they should be hosted in IIS.

Comment: but why IIS???

File: docs/connect-storefront-to-platform-v3.md [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+ "Endpoint": {
+     //Specify platform url
+     "Url": "https://localhost:5001",
+     //Comment the follow settings
+     //"AppId": "...",
+     //"SecretKey":"...",
+
+     //Uncomment the follow settings
+      "UserName": "admin",
+      "Password": "store",
```
Comment (eMazeika) [inline docs/connect-storefront-to-platform-v3.md]: switch the "Comment" with "Uncomment"??

File: docs/connect-storefront-to-platform-v3.md [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+      "RequestTimeout": "0:0:30"
+    },
+```
+
+* Trust the .Net Core Development Self-Signed Certificate. Example how to trust a self-signed certificate you can find in this [article](https://blogs.msdn.microsoft.com/robert_mcmurray/2013/11/15/how-to-trust-the-iis-express-self-signed-certificate/)
```
Comment (eMazeika) [inline docs/connect-storefront-to-platform-v3.md]: More details on trusting the self-signed certificate can be found [here]

File: docs/deploy-from-precompiled-binaries-windows.md (hunk 6 above)
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-windows.md]: fixed

File: docs/connect-storefront-to-platform-v3.md (hunk 13 above)
Comment (mvktsk) [reply docs/connect-storefront-to-platform-v3.md]: fixed

File: docs/connect-storefront-to-platform-v3.md (hunk 14 above)
Comment (mvktsk) [reply docs/connect-storefront-to-platform-v3.md]: Note, if Platform and Storefront are deployed in the same on-premises environment, Storefront should be deployed on different port then Platform. You can do it by `dotnet run CLI`

```console
dotnet VirtoCommerce.Storefront.dll --urls=http://localhost:5002/
```

or host Storefront in IIS.

File: docs/connect-storefront-to-platform-v3.md (hunk 15 above)
Comment (mvktsk) [reply docs/connect-storefront-to-platform-v3.md]: User name and password used to connect to Platform V3

"UserName": "admin",
"Password": "store",
lines commented  by default in Storefront appsettings.json

File: docs/connect-storefront-to-platform-v3.md (hunk 16 above)
Comment (mvktsk) [reply docs/connect-storefront-to-platform-v3.md]: fixed

File: docs/deploy-from-precompiled-binaries-linux.md [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+            "PublicUrl": "https://localhost:5001/assets/" <-- Set your platform application url with port localhost:5001
+        },
+    },
+```
+
+### Running the Platform by CLI "dotnet"
+
+* Install and trust HTTPS certificate
+
+Run steps described in this article[article](https://docs.microsoft.com/en-us/aspnet/core/security/enforcing-ssl?view=aspnetcore-3.0&tabs=visual-studio#trust-https-certificate-from-windows-subsystem-for-linux) to trust the .NET Core SDK HTTPS development certificate on Linux.
```
Comment (eMazeika) [inline docs/deploy-from-precompiled-binaries-linux.md]: in [this article]

File: docs/deploy-from-precompiled-binaries-linux.md [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+The output in the console will say something like:
+
+```console
+Now listening on: http://localhost:5000
+Now listening on: https://localhost:5001
+```
+
+### First run sign in
```
Comment (eMazeika) [inline docs/deploy-from-precompiled-binaries-linux.md]: First run and sign in

File: docs/deploy-from-precompiled-binaries-windows.md [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+
+* Open `https://localhost:5001` url in your browser. "Your connection is not private" might appear. Click "Advanced" and "Proceed to ...".
+Read more on removing this error and using a self-signed certificate: [Trust the ASP.NET Core HTTPS development certificate](https://www.hanselman.com/blog/DevelopingLocallyWithASPNETCoreUnderHTTPSSSLAndSelfSignedCerts.aspx)
+* The application will create and initialize database on the first request. After that you should see the sign in page. Use the following credentials:
+  * Login: **admin**
+  * Password: **store**
+
+### Host on Windows with IIS
+
+VirtoCommerce.Platform project already include the **web.config** ffile is already included in the release package. It contains all the necessary settings for running in IIS.
```
Comment (eMazeika) [inline docs/deploy-from-precompiled-binaries-windows.md]: The required **web.config** file was already included in the release package.

File: docs/deploy-from-precompiled-binaries-linux.md (hunk 17 above)
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-linux.md]: fixed

File: docs/deploy-from-precompiled-binaries-linux.md (hunk 18 above)
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-linux.md]: fixed

File: docs/deploy-from-precompiled-binaries-windows.md (hunk 19 above)
Comment (mvktsk) [reply docs/deploy-from-precompiled-binaries-windows.md]: fixed

**Pull request merged**